### PR TITLE
Add the missing file operations and add the path manipulation

### PR DESCRIPTION
**Pull request by TeoDev1611**
Hello

I added the delete operation and the path manipulation:

- The delete explain how delete a empty and a non empty directory synchronously and async.
- The path manipulation explain how join, the base name, the file from url, the current directory and format the path 👍🏼 

I hope to help you

### CODE REVIEW

File: data/deleting-files.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
+ * @title Deleting files
+ * @difficulty beginner
+ * @tags cli
+ * @run --allow-write <url>
+ * @resource {https://doc.deno.land/deno/stable/~/Deno.remove} Doc: Deno.remove
+ * @resource {https://doc.deno.land/deno/stable/~/Deno.removeSync} Doc: Deno.removeSync
+ *
+ * Many applications need to delete files from disk. Deno provides a simple
+ * interface for deleting files.
```
Comment (lucacasonato) [inline data/deleting-files.ts:10]:
```suggestion
 * Deleting files from disk is a common operation. Deno provides a simple
 * interface for deleting files.
```

File: data/deleting-files.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+// The easiest way to delete a file or directory empty, is pass the path and
+// delete this file.
```
Comment (lucacasonato) [inline data/deleting-files.ts:14]:
```suggestion
// Files and empty directories can be deleted with the `Deno.remove` API.
```

File: data/deleting-files.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/**
+ * @title Deleting files
```
Comment (lucacasonato) [inline data/deleting-files.ts:2]:
```suggestion
 * @title Deleting files and directories
```

File: data/deleting-files.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * Many applications need to delete files from disk. Deno provides a simple
+ * interface for deleting files.
+ */
+
+// The easiest way to delete a file or directory empty, is pass the path and
+// delete this file.
+await Deno.remove("/path/to/empty_dir/or/file");
+
+// Throws error if permission is denied, path not found, or path is non empty
+// directory for fix the last error you need set recursive true
```
Comment (lucacasonato) [inline data/deleting-files.ts:18]:
```suggestion
// To delete a directory with its contents, set the `recursive` flag on the
// Deno.remove call.
```

File: data/deleting-files.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+await Deno.remove("/path/to/populated_dir/or/file", { recursive: true });
+
+// For remove the file or directory synchronously you can use this:
+Deno.removeSync("/path/to/empty_dir/or/file");
+
+// For remove a non empty directorty synchronously you set recursive true
+Deno.removeSync("/path/to/populated_dir/or/file", { recursive: true });
```
Comment (lucacasonato) [inline data/deleting-files.ts:25]:
```suggestion
// A synchronous version of the API is also available.
Deno.removeSync("/path/to/empty_dir/or/file");
```

File: data/deleting-files.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+/**
+ * @title Deleting files
+ * @difficulty beginner
+ * @tags cli
+ * @run --allow-write <url>
```
Comment (lucacasonato) [inline data/deleting-files.ts:5]:
```suggestion
```

**Review by lucacasonato — COMMENTED** [review]
Thanks for taking the time to contribute. I have not yet reviewed the path-operations example. I'll do that soon.

File: data/path-operations.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+ * interface for deleting files.
+ */
+
+// First well import the librery from the Deno Std
+import * as path from "https://deno.land/std@0.120.0/path/mod.ts";
+
+// If you want from url file to a os dir you can use this
+const p1 = path.posix.fromFileUrl("file:///home/foo");
+const p2 = path.win32.fromFileUrl("file:///home/foo");
+console.log(`PATH 1: ${p1} PATH 2: ${p2}`); // PATH 1: /home/foo PATH 2: \\home\\foo
```
Comment (lino-levan) [inline data/path-operations.ts:19]:
```suggestion
// Converting from a file url to a directory can be done simply by the `fromFileUrl`
// method from the appropriate implementation.
const p1 = path.posix.fromFileUrl("file:///home/foo");
const p2 = path.win32.fromFileUrl("file:///home/foo");
console.log(`PATH 1: ${p1} PATH 2: ${p2}`); // PATH 1: /home/foo PATH 2: \\home\\foo

// We can also choose to not specify and automatically use whatever Deno is running on
const p3 = path.fromFileUrl("file:///home/foo");
```

File: data/path-operations.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+// You want get the last part of a file path you can use this
```
Comment (lino-levan) [inline data/path-operations.ts:21]:
```suggestion
// We can get the last part of a file path using the basename method
```

File: data/path-operations.ts [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+const p = path.basename("./deno/is/awesome/mod.ts");
+console.log(p); // mod.ts
+
+// You want get the directory name from a Path
```
Comment (lino-levan) [inline data/path-operations.ts:25]:
```suggestion
// We can get the directory of a file path using the dirname method
```

File: data/path-operations.ts [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+const base = path.dirname("./deno/is/awesome/mod.ts");
+console.log(base); // ./deno/is/awesome
+
+// You want get the extension from a path
```
Comment (lino-levan) [inline data/path-operations.ts:29]:
```suggestion
// We can get the extension of a file path using the extname method
```

File: data/path-operations.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+const ext = path.extname("./deno/is/awesome/mod.ts");
+console.log(ext); // .ts
+
+// You want format a path generating from a object
```
Comment (lino-levan) [inline data/path-operations.ts:33]:
```suggestion
// We can format a path using a FormatInputPathObject
```

File: data/path-operations.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+const formatPath = path.format({
+  root: "/",
+  dir: "/home/user/dir",
+  ext: ".html",
+  name: "index",
+});
+console.log(formatPath); // "/home/user/dir/index.html"
+
+// You want join files with the os separator
```
Comment (lino-levan) [inline data/path-operations.ts:42]:
```suggestion
// When we want to make our code cross-platform, we can use the join method.
// This joins any number of string by the OS-specific file seperator
```
We should probably move this up the file as it is probably the most important utility.

File: data/path-operations.ts [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+const joinPath = path.join("foo", "bar");
+console.log(joinPath); // foo/bar LINUX MACOS -- foo\\bar WINDOWS
+
+// You want get the current directory
```
Comment (lino-levan) [inline data/path-operations.ts:46]:
```suggestion
// We can get the current working directory using the built-in cwd method
```

**Review by lino-levan — CHANGES_REQUESTED** [review]
A lot of good work, but somewhat outdated. Please fix these (or just accept the suggested changes).

**Comment by lino-levan** [comment]
@TeoDev1611 are you willing to move this forward?

**Comment by lino-levan** [comment]
@TeoDev1611 glad to see you moving this forward. Besides my two remaining comments, this LGTM!

Pinging @crowlKats to please take a look!

**Comment by TeoDev1611** [comment]
> @TeoDev1611 glad to see you moving this forward. Besides my two remaining comments, this LGTM!
> 
> Pinging @crowlKats to please take a look!

Thanks for the suggestions, I incorporated all the changes!

**Comment by lino-levan** [comment]
Perfect! LGTM.

**Comment by crowlKats** [comment]
could you please run deno fmt?

File: data/path-operations.ts [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+// We can get the last part of a file path using the basename method
+const p = path.basename("./deno/is/awesome/mod.ts");
+console.log(p); // mod.ts
+
+// We can get the directory of a file path using the dirname method
+const base = path.dirname("./deno/is/awesome/mod.ts");
+console.log(base); // ./deno/is/awesome
+
+// You want get the extension from a path
```
Comment (lino-levan) [inline data/path-operations.ts:33]:
```suggestion
// We can get the extension of a file path using the extname method
```

**Comment by lino-levan** [comment]
@hashrock I believe this should be closed now that #62 got merged

**Comment by hashrock** [comment]
Yeah! thank you for your contribution @TeoDev1611 !

**Pull request closed without merging**